File: android/app/src/main/kotlin/com/example/ultra_mobile/TodoNativeApiImpl.kt
package com.example.ultra_mobile

import com.example.ultra_mobile.TodoDetailNative
import com.example.ultra_mobile.TodoNativeApi

class TodoNativeApiImpl : TodoNativeApi {
    override fun fetchTodoDetail(id: Long, callback: (kotlin.Result<TodoDetailNative>) -> Unit) {
        // In a real app, this might fetch from a native database or another service.
        // For this challenge, we simulate providing extended details for the Task.
        
        val detail = TodoDetailNative(
            id = id,
            description = "This is a detailed description for task #$id, provided natively by Android. It includes more context than the simple task title seen on the main list.",
            category = if (id % 2 == 0L) "Work" else "Personal",
            priority = (id % 3).toLong(),
            dueDate = System.currentTimeMillis() + (86400000 * 7), // Due in 7 days
            imageUrl = "https://picsum.photos/seed/$id/600/400"
        )
        
        callback(kotlin.Result.success(detail))
    }
}

File: ios/Runner/TodoNativeApiImpl.swift
import Foundation
import Flutter

class TodoNativeApiImpl: NSObject, TodoNativeApi {
    func fetchTodoDetail(id: Int64, completion: @escaping (Result<TodoDetailNative, Error>) -> Void) {
        // Mocking some extended data based on the ID
        // In a real app, this might come from a local database or a platform-specific storage
        
        let detail = TodoDetailNative(
            id: id,
            description: "Detailed description for task \(id) fetched from native iOS bridge.",
            category: id % 2 == 0 ? "Work" : "Personal",
            priority: Int64(id % 3), // mapped to Priority enum in Flutter: 0: High, 1: Medium, 2: Low
            dueDate: Int64(Date().addingTimeInterval(86400 * 7).timeIntervalSince1970 * 1000), // 7 days from now
            imageUrl: "https://picsum.photos/id/\(id % 100)/400/200"
        )
        
        // Simulating a slight delay to mimic network or database fetch
        DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
            completion(.success(detail))
        }
    }
}
